backtrace-library/src/main/java/backtraceio/library/common/BatteryStateHelper.java

```diff
@@ -0,0 +1,198 @@
+package backtraceio.library.common;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.BatteryManager;
+
+
+import java.util.HashMap;
+
+import backtraceio.library.BacktraceDatabase;
+import backtraceio.library.interfaces.Database;
+
+public class BatteryStateHelper extends BroadcastReceiver {
+    private static Context context;
+    private static BacktraceDatabase database;
+    private static boolean trackBattery;
+    private static String batteryHealth, batteryLevel, batteryChargingSource, batteryChargingStatus, batteryTechnology, batteryTemperature;
+
+    public BatteryStateHelper(Context context) {
+        this(context, null);
+    }
+
+    public static HashMap<String, String> getValues() {
+        HashMap<String, String> result = new HashMap<String, String>();
+        if (!trackBattery) {
+            result.put("battery.health", "N/A");
+            result.put("battery.level", "N/A");
+            result.put("battery.charging.source", "N/A");
+            result.put("battery.charging.status", "N/A");
+            result.put("battery.technology", "N/A");
+            result.put("battery.temperature", "N/A");
+            return result;
+        }
+        result.put("battery.health", batteryHealth);
+        result.put("battery.level", batteryLevel);
+        result.put("battery.charging.source", batteryChargingStatus);
+        result.put("battery.charging.status", batteryChargingSource);
+        result.put("battery.technology", batteryTechnology);
+        result.put("battery.temperature", batteryTemperature);
+        return result;
+    }
+
+    public BatteryStateHelper(Context context, Database database) {
+        if (BatteryStateHelper.context == null){
+            BatteryStateHelper.context = context;
+            if (database != null) {
+                BatteryStateHelper.database = (BacktraceDatabase)database;
+            }
+        }
+    }
+
+    public void disable() {
+        trackBattery = false;
+    }
+
+    public void enable() {
+        trackBattery = true;
+        IntentFilter ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);

```

Comment:
Why do we need to register an intent filter for the same action twice? Maybe we can remove this redundancy by doing all the `context.registerReceiver` calls from inside here when we move them out of `BacktraceClient`.